### ISSUE TITLE
get_associated_token_address() parameters reversed from solana/web3.js

### ISSUE THREAD

**Issue by clmentcoutet** [michaelhly/solana-py#470]
In this function `get_associated_token_address(owner: Pubkey, mint: Pubkey, token_program_id: Pubkey = TOKEN_PROGRAM_ID)`, the parameters are inversed (mint and owner) compared to the exact same function from solana/web.js
`export function getAssociatedTokenAddressSync(
    mint: PublicKey,
    owner: PublicKey,
    allowOwnerOffCurve = false,
    programId = TOKEN_PROGRAM_ID,
    associatedTokenProgramId = ASSOCIATED_TOKEN_PROGRAM_ID
)`
This is very confusing and lead to mistake when trying to replicate JS example in Python

**Comment by kingsznhone** [comment]
Different Solana client libraries (web3.js, solana-py, solana-kit, etc.) expose different API shapes and parameter orders, but they are ultimately just thin abstractions over the same deterministic PDA derivation logic.

For example, Associated Token Accounts are not “looked up” via RPC—they are derived via find_program_address with a fixed seed layout defined by the SPL Associated Token Program.

As a result:

The input structure differs per SDK (ordering, types, optional flags)
But the underlying derivation formula is identical

In other words, these libraries are different “dialects” over the same canonical algorithm. The output is deterministic as long as:

the same mint
the same owner
the same token/ATA program IDs
are used.

So differences between web3.js / solana-kit / solana-py are purely ergonomic/API-level, not semantic.

It is recommend use `get_associated_token_address(owner=wallet, mint=quote_mint)` instead of rely on args sequences.
